src/usb_reset/src/ros_usb_reset.cpp

```diff
@@ -0,0 +1,89 @@
+#include "usb_reset/ros_usb_reset.hpp"
+
+int num = 0;
+
+ROS_Usb_Reset::ROS_Usb_Reset(QQuickItem * parent) :
+    QQuickPaintedItem(parent),
+    topic_value(),
+    ros_ready(false),
+    devices("no devices") {
+
+}
+
+void ROS_Usb_Reset::setup(ros::NodeHandle * nh, QQmlContext *ctxt) {
+
+    //TODO: make these parameters of the component
+      usb_sub = nh->subscribe(
+            topic_value.toStdString(),
+            1,
+            &ROS_Usb_Reset::receive_msg,
+            this
+      );
+      usb_pub = nh->advertise<std_msgs::Int16>(
+                    "rover/usb/reset",
+                    1,
+                    false
+            );
+
+    ros_ready = true;
+    curr_ctxt = ctxt;
+    ROS_INFO("Setup of usb component complete");
+}
+
+void ROS_Usb_Reset::receive_msg(const std_msgs::String::ConstPtr &msg) {
+    devices = msg->data;
+    update();
+}
+
+void ROS_Usb_Reset::paint(QPainter * painter) {
+  QLinearGradient linearGradient(0, 0, 100, 100);
+  linearGradient.setColorAt(0.2, Qt::yellow);
+  painter->setBrush(linearGradient);
+  painter->setPen(Qt::green);
+  int i = 0;
+  QString dev = QString::fromStdString(devices);
+  QStringList list = dev.split("#/#/");
+  //create the neccessary number of buttons
+  curr_ctxt->setContextProperty("myModel", (list.size() - 1));
```

Comment:
please name this model something like `usbDevices` so we don't misuse the namespace